contracts/extensions/ExpenditureUtils.sol

```diff
@@ -0,0 +1,318 @@
+  }
+
+  function reclaimStake(uint256 _expenditureId) public {
+    Stake storage stake = stakes[_expenditureId];
+    require(stake.amount > 0, "expenditure-utils-nothing-to-claim");
+
+    ColonyDataTypes.Expenditure memory expenditure = colony.getExpenditure(_expenditureId);
+    require(
+      expenditure.status == ColonyDataTypes.ExpenditureStatus.Cancelled ||
+      expenditure.status == ColonyDataTypes.ExpenditureStatus.Finalized,
+      "expenditure-utils-expenditure-invalid-state"
+    );
+
+    colony.deobligateStake(stake.creator, expenditure.domainId, stake.amount);
+
+  // slither-disable-next-line reentrancy-no-eth
+    delete stakes[_expenditureId];
+  }
+
+  function cancelAndReclaimStake(
+    uint256 _permissionDomainId,
+    uint256 _childSkillIndex,
+    uint256 _expenditureId
+  )
+    public
+  {
+    Stake storage stake = stakes[_expenditureId];
+    ColonyDataTypes.Expenditure memory expenditure = colony.getExpenditure(_expenditureId);
+
+    require(expenditure.owner == msgSender(), "expenditure-utils-must-be-owner");
+
+    require(
+      expenditure.status == ColonyDataTypes.ExpenditureStatus.Draft,
+      "expenditure-utils-expenditure-not-draft"
+    );
+
+    cancelExpenditure(_permissionDomainId, _childSkillIndex, _expenditureId, expenditure.owner);
+    reclaimStake(_expenditureId);
+  }
+
+  function cancelAndPunish(
+    uint256 _permissionDomainId,
+    uint256 _childSkillIndex,
+    uint256 _expenditureId,
+    bool _punish
+  )
+    public
+  {
+    ColonyDataTypes.Expenditure memory expenditure = colony.getExpenditure(_expenditureId);
+
+    require(
+      colony.hasInheritedUserRole(msgSender(), _permissionDomainId, ColonyDataTypes.ColonyRole.Arbitration, _childSkillIndex, expenditure.domainId),
+      "expenditure-utils-caller-not-arbitration"
+    );
+
+    require(
+      expenditure.status != ColonyDataTypes.ExpenditureStatus.Draft,
+      "expenditure-utils-expenditure-still-draft"
+    );
+
+    if (_punish) {
+      Stake storage stake = stakes[_expenditureId];
+      require(stake.amount > 0, "expenditure-utils-nothing-to-slash");
+
+      colony.transferStake(_permissionDomainId, _childSkillIndex, address(this), stake.creator, expenditure.domainId, stake.amount, address(0x0));
+
+      colony.emitDomainReputationPenalty(
+        _permissionDomainId,
+        _childSkillIndex,
+        expenditure.domainId,
+        stake.creator,
+        -int256(stake.amount)
+      );
+
+      // slither-disable-next-line reentrancy-no-eth
+      delete stakes[_expenditureId];
+    }
+
+    cancelExpenditure(_permissionDomainId, _childSkillIndex, _expenditureId, expenditure.owner);
+  }
+
+  uint256 constant EXPENDITURE_SLOT = 25;
+  uint256 constant EXPENDITURESLOTS_SLOT = 26;
+  uint256 constant PAYOUT_MODIFIER_OFFSET = 2;
+  bool constant MAPPING = false;
+  bool constant ARRAY = true;
+
+  /// @notice Sets the payout modifiers in given expenditure slots, using the arbitration permission
```

Comment:
This is quite misleading - it does use the arbitration permission (of the extension), but is restricted to the owner of the expenditure. 

Separately, what's the requirement this function is allowing us to meet? I'm not seeing it.

Reply:
It's not a part of this PR, it was introduced back in https://github.com/JoinColony/colonyNetwork/pull/958

The idea was that we only wanted payout modifiers to be set by arbitration users on the core contracts, but provide colonies a way to allow owners to set payout modifiers via an optional extension.

Reply:
So it should be its own extension, or toggleable, or something? Because a colony might want to enable salaries, but not the functionality you've described?

Reply:
That's a fair point -- I had initially thought to bundle the functionality together, but keeping them separate makes sense.